### PR TITLE
Bug 1103810 - make 4 call forwarding options shown as dialog

### PR DESCRIPTION
**Pull request by EragonJ**
... shown as a dialog

### CODE REVIEW

**Comment by try-server-hook** [comment]
[EragonJ EragonJ (E.J.) (EragonJ)](https://github.com/EragonJ) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=5da757b42f0b)

File: apps/settings/elements/call_cf_mobile_busy_settings.html [hunk 1]
```diff
@@ -1,27 +1,29 @@
...
-            <span data-l10n-id="callForwardingForward">Forward</span>
-            <small data-l10n-id="callForwardingMobileBusyShort">When busy</small>
+            <span data-l10n-id="callForwardingForward"></span>
+            <small data-l10n-id="callForwardingMobileBusyShort"></small>
           </label>
         </li>
       </ul>
     </div>
 
+    <panel data-path="panels/call_cf/panel"></panel>
```
Comment (crh0716) [inline apps/settings/elements/call_cf_mobile_busy_settings.html]: nit: Let's keep the module in the `panels/call_forwarding_details/` folder. (Save `panels/call_forwarding` for the "Call Forwarding" panel.)

File: apps/settings/js/call.js [hunk 2]
```diff
@@ -491,66 +490,51 @@ require([
...
+        if (!cs_isPhoneNumberValid(number)) {
+          DialogService.alert('callForwardingInvalidNumberError', {
+            title: 'callForwardingConfirmTitle',
+            submitButtonText: 'continue'
+          });
+          cs_enableTapOnCallerIdItem(false);
+          cs_enableTapOnCallWaitingItem(false);
+          cs_enableTapOnCallForwardingItems(false);
+          cs_updateCallForwardingSubpanels();
+          return;
```
Comment (crh0716) [inline apps/settings/js/call.js]: We can remove the return as you already add a `else` case.

File: apps/settings/elements/call_cf_mobile_busy_settings.html (hunk 1 above)
Comment (EragonJ) [inline apps/settings/elements/call_cf_mobile_busy_settings.html]: ok !

File: apps/settings/js/call.js (hunk 2 above)
Comment (EragonJ) [inline apps/settings/js/call.js]: ok !!

**Comment by try-server-hook** [comment]
[EragonJ EragonJ (E.J.) (EragonJ)](https://github.com/EragonJ) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=f93938a4880b)

**Pull request merged**